### PR TITLE
Fix NPE in DiscoveryManager after restarting Okapi OKAPI-976

### PR DESCRIPTION
**Pull request by adamdickmeiss**

### CODE REVIEW

**Review by hjiebsco — APPROVED** [review]
The new `addAndDeployIgnoreError` method can address the NPE when deploying to an outdated node. What about other potential cases when the `nodes` async map and the actual cluster nodes are out of sync? Should those be failed or ignored as well?

File: okapi-core/src/main/java/org/folio/okapi/managers/DiscoveryManager.java [hunk 1]
```diff
@@ -490,7 +483,15 @@ private boolean isAlive(DeploymentDescriptor md, Collection<NodeDescriptor> node
   }
 
   Future<NodeDescriptor> getNode(String nodeId) {
-    return nodeUrl(nodeId).compose(x -> getNode1(x));
+    return nodeUrl(nodeId)
```
[the remark this answers is not in the document]
Comment (adamdickmeiss) [reply okapi-core/src/main/java/org/folio/okapi/managers/DiscoveryManager.java:486]: Yes. Renamed to nodeUrlToId

File: okapi-core/src/main/java/org/folio/okapi/managers/DiscoveryManager.java [hunk 2]
```diff
@@ -490,7 +483,15 @@ private boolean isAlive(DeploymentDescriptor md, Collection<NodeDescriptor> node
...
+        .compose(nodeDescriptor -> getNode1(nodeDescriptor))
```
[the remark this answers is not in the document]
Comment (adamdickmeiss) [reply okapi-core/src/main/java/org/folio/okapi/managers/DiscoveryManager.java:487]: Right. Renamed and calling the main argument for nodeName (informal name)

File: okapi-core/src/test/java/org/folio/okapi/managers/DiscoveryManagerTest.java [hunk 3]
```diff
@@ -13,18 +13,32 @@
...
+  public void after(TestContext context) {
+    vertx.close(context.asyncAssertSuccess());
+  }
+
   @Test
   public void isLeaderWithoutClusterManager(TestContext context) {
     DiscoveryManager discoveryManager = new DiscoveryManager(null);
 
-    discoveryManager.init(Vertx.vertx()).onComplete(context.asyncAssertSuccess(then ->
+    discoveryManager.init(vertx).onComplete(context.asyncAssertSuccess(then ->
```
[the remark this answers is not in the document]
Comment (adamdickmeiss) [reply okapi-core/src/test/java/org/folio/okapi/managers/DiscoveryManagerTest.java:41]: Yes. Missed those.

**Comment by adamdickmeiss** [comment]
I think all errors regarding auto-deployment should be ignored.. Eg if an instance is no longer available or can not be launched, that should not fail a start of Okapi.. Just like if Okapi Øis* running already, a deployment failure should of course not make Okapi stop and abort.

**Comment by adamdickmeiss** [comment]
> The new `addAndDeployIgnoreError` method can address the NPE when deploying to an outdated node. What about other potential cases when the `nodes` async map and the actual cluster nodes are out of sync? Should those be failed or ignored as well?

I think all errors regarding auto-deployment should be ignored.. Eg if an instance is no longer available or can not be launched, that should not fail a start of Okapi.. Just like if Okapi is running already, a deployment failure should of course not make Okapi stop and abort.

**Pull request merged**